events/include/events/EventQueue.h

```diff
@@ -89,30 +89,28 @@ class EventQueue : private mbed::NonCopyable<EventQueue> {
     /** Dispatch events
      *
      *  Executes events until the specified milliseconds have passed.
-     *  If ms is negative, the dispatch function will dispatch events
-     *  indefinitely or until break_dispatch is called on this queue.
      *
-     *  When called with a finite timeout, the dispatch function is guaranteed
-     *  to terminate. When called with a timeout of 0, the dispatch function
-     *  does not wait and is IRQ safe.
+     *  The dispatch function is guaranteed to terminate after the elapsed wait
      *
-     *  @param ms       Time to wait for events in milliseconds, a negative
-     *                  value will dispatch events indefinitely
-     *                  (default to -1)
+     *  @param wait     Time to wait for events in milliseconds, expressed as a 
+     *                  Chrono duration.
      */
-    void dispatch(int ms = -1);
+    void dispatch(duration wait);
 
     /** Dispatch events without a timeout
-     *
-     *  This is equivalent to EventQueue::dispatch with no arguments, but
-     *  avoids overload ambiguities when passed as a callback.
-     *
-     *  @see EventQueue::dispatch
+     * 
+     * Executes events indefinitely unless the dispatch loop is forcibly broken.
+     * See break_dispatch()
+     * 
      */
-    void dispatch_forever()
-    {
-        dispatch();
-    }
+    void dispatch_forever();
+
+    /** Dispatch currently queued events only and then terminate
+     * 
+     * In this case, the dispatch function does not wait and is IRQ safe.
```

Comment:
I think we can't quite say it's IRQ safe. It wouldn't be in an RTOS build, as it still uses mutexes for its state locks. It's close to it, and in certain specialised non-RTOS runtime environments this might be important, but for mainline Mbed OS API I think I'd like to not try to specify that.